Xamarin.Forms.Platform.Android/CollectionView/ItemsViewRenderer.cs

```diff
@@ -138,7 +138,13 @@ protected override void Dispose(bool disposing)
 				if (Element != null)
 				{
 					TearDownOldElement(Element as ItemsView);
+				}
+
+				// Dispose the adapter after tear down element
+				ItemsViewAdapter?.Dispose();
```

Comment:
`TearDownOldElement` already disposes of whichever adapter was active at the time it was called. So if ItemsViewAdapter is the active adapter when `TearDownOldElement` is called, it will be disposed.

I think the problem might occur if `_emptyViewAdapter` is the active adapter when disposing - then `_emptyViewAdapter` will be disposed, but `ItemsViewAdapter` will not. 

So I think the problem is in `TearDownOldElement` - it should be cleaning up both adapters.

Reply:
I'm sure it's not the `EmptyViewAdapter` the problem because from the stacktrace of the crash, it  is caused by the event of the ItemSource (`ObservableItemsSource:CollectionChanged` which is not unsubscribed. And this `ItemSource` is in the `ItemsViewAdapter` and not in the EmptyViewAdapter.

So the problem is because an `ItemsViewAdapter` is not disposed.
This indeed caused by the `EmptyViewAdapter` because if the `GetAdapter` method return the `EmptyViewAdapter` the `ItemsViewAdapter` will not be disposed and cause the crash.

I will made some more tests to make a more robust fix.